torchgeo/datasets/eurocrops.py

```diff
@@ -0,0 +1,285 @@
+    publicly available self-declared crop reporting datasets from countries of the
+    European Union" into a unified format. The dataset is released under CC BY 4.0 Deed.
+
+    The dataset consists of shapefiles containing a total of X polygons. Each polygon
+    is tagged with a "EC_hcat_n" attribute indicating the harmonized crop name grown
+    within the polygon in the year associated with the shapefile.
+
+    If you use this dataset in your research, please follow the citation guidelines at
+    https://github.com/maja601/EuroCrops#reference.
+    """
+
+    base_url = "https://zenodo.org/records/8229128/files/"
+
+    hcat_fname = "HCAT2.csv"
+    hcat_md5 = "2443cb1ef05ad16840f0f6a4c685b72d"
+    # Valid HCAT levels.
+    hcat_levels = [3, 4, 5, 6]
+    # Name of the column containing HCAT code in CSV file.
+    hcat_code_column = "HCAT2_code"
+
+    label_name = "EC_hcat_c"
+
+    # Filename, year, and md5 of files in this dataset on zenodo.
+    zenodo_files = [
+        ("AT_2021.zip", 2021, "490241df2e3d62812e572049fc0c36c5"),
+        ("BE_VLG_2021.zip", 2021, "ac4b9e12ad39b1cba47fdff1a786c2d7"),
+        ("DE_LS_2021.zip", 2021, "6d94e663a3ff7988b32cb36ea24a724f"),
+        ("DE_NRW_2021.zip", 2021, "a5af4e520cc433b9014cf8389c8f4c1f"),
+        ("DK_2019.zip", 2019, "d296478680edc3173422b379ace323d8"),
+        ("EE_2021.zip", 2021, "a7596f6691ad778a912d5a07e7ca6e41"),
+        ("ES_NA_2020.zip", 2020, "023f3b397d0f6f7a020508ed8320d543"),
+        ("FR_2018.zip", 2018, "282304734f156fb4df93a60b30e54c29"),
+        ("HR_2020.zip", 2020, "8bfe2b0cbd580737adcf7335682a1ea5"),
+        ("LT_2021.zip", 2021, "c7597214b90505877ee0cfa1232ac45f"),
+        ("LV_2021.zip", 2021, "b7253f96c8699d98ca503787f577ce26"),
+        ("NL_2020.zip", 2020, "823da32d28695b8b016740449391c0db"),
+        ("PT.zip", 2021, "3dba9c89c559b34d57acd286505bcb66"),
+        ("SE_2021.zip", 2021, "cab164c1c400fce56f7f1873bc966858"),
+        ("SI_2021.zip", 2021, "6b2dde6ba9d09c3ef8145ea520576228"),
+        ("SK_2021.zip", 2021, "c7762b4073869673edc08502e7b22f01"),
+        # Year is unknown for Romania portion (ny = no year).
+        # ("RO_ny", ??),
+    ]
+
+    # Color palette to choose from.
+    # There are hundreds of classes so we pick color via modulo of class index.
+    colors = [
+        (70, 107, 159, 255),
+        (209, 222, 248, 255),
+        (222, 197, 197, 255),
+        (217, 146, 130, 255),
+        (235, 0, 0, 255),
+        (171, 0, 0, 255),
+        (179, 172, 159, 255),
+        (104, 171, 95, 255),
+        (28, 95, 44, 255),
+        (181, 197, 143, 255),
+        (204, 184, 121, 255),
+        (223, 223, 194, 255),
+        (220, 217, 57, 255),
+        (171, 108, 40, 255),
+        (184, 217, 235, 255),
+        (108, 159, 184, 255),
+    ]
+
+    def __init__(
+        self,
+        paths: Union[str, Iterable[str]] = "data",
+        crs: Optional[CRS] = None,
+        res: float = 0.00001,
```

Comment:
This may need to be changed depending on the units of the CRS. A default with a similar resolution as Sentinel-2 would make sense to me.

Reply:
The files range in CRS from WGS-84 to EPSG:31287 so I have set a default CRS (WGS-84), with res equivalent to 10 m/pixel at equator.